src/halmos/__main__.py

```diff
@@ -1193,6 +1193,11 @@ def run_test(ctx: FunctionContext) -> TestResult:
             REVERT_ALL,
             f"{funsig}: all paths have been reverted; the setup state or inputs may have been too restrictive.",
         )
+    elif len(ctx.solver_outputs) < len(submitted_futures):
```

Comment:
I'm a bit concerned about relying on the number of solver outputs to infer that an error occured

> when an exception is raised in solver subprocesses, their result may not be added to the solver_outputs list

this seems to be the real issue? ideally we want to capture the errors and add failed outputs to `ctx.solver_outputs`, no?

Reply:
> ideally we want to capture the errors and add failed outputs to ctx.solver_outputs, no?

yes, but it needs some more work, as the solver output object currently needs more info that isn't available at some places exceptions occur.